cdap-notifications/src/main/java/co/cask/cdap/notifications/service/inmemory/InMemoryNotificationService.java

```diff
@@ -64,6 +65,10 @@ protected void shutDown() throws Exception {
   public <N> ListenableFuture<N> publish(final Id.NotificationFeed feed, final N notification,
                                          final Type notificationType)
     throws NotificationException {
+    if (executorService == null) {
```

Comment:
It should never happen because if `executorService` is null, meaning this `InMemoryNotificationService`, which is a guava service, is not started. Technically it should be an IllegalStateException.